### PR TITLE
Filter out possibly stuck relocations on SearchableSnapshotsRelocationIntegTests.testRelocationWaitsForPreWarm

### LINKED ISSUE #70627: [CI] SearchableSnapshotsRelocationIntegTests.testRelocationWaitsForPreWarm fails
**Build scan**:

This PR test run: https://gradle-enterprise.elastic.co/s/3amvbmcuylppi
But also earlier runs on CI, e.g. https://gradle-enterprise.elastic.co/s/l7ezvyjdj4u5s

**Repro line**:

./gradlew ':x-pack:plugin:searchable-snapshots:internalClusterTest' --tests "org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsRelocationIntegTests.testRelocationWaitsForPreWarm" -Dtests.seed=BB10EA14543E1244 -Dtests.security.manager=true -Dtests.locale=fr-FR -Dtests.timezone=Europe/Oslo -Druntime.java=11

**Reproduces locally?**:

Yes

**Applicable branches**:

master, some earlier failures on 7.x

**Failure history**:

Found three failures in Dec. 2020

**Failure excerpt**:

```
 java.lang.AssertionError:
    Expected: an empty collection
         but: <[org.elasticsearch.indices.recovery.RecoveryState@9038606]>
        at __randomizedtesting.SeedInfo.seed([BB10EA14543E1244:C4CC9B2A028D3DAA]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.xpack.searchablesnapshots.SearchableSnapshotsRelocationIntegTests.lambda$testRelocationWaitsForPreWarm$3(SearchableSnapshotsRelocationIntegTests.java:114)
```

### PR DESCRIPTION
**Pull request by fcofdez**
Closes #70627
Relates #70311

### CODE REVIEW

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java [hunk 1]
```diff
@@ -114,7 +115,19 @@ public void testRelocationWaitsForPreWarm() throws Exception {
...
 
+    private static List<RecoveryState> getRelocations(String restoredIndex) {
+        return getRelocationsStream(restoredIndex).collect(Collectors.toList());
+    }
+
     private static List<RecoveryState> getActiveRelocations(String restoredIndex) {
+        return getRelocationsStream(restoredIndex)
+            // filter for relocations that are not in stage FINALIZE (they could end up in this stage without progress for good if the
+            // target node does not have enough cache space available to hold the primary completely
+            .filter(recoveryState -> recoveryState.getStage() != RecoveryState.Stage.FINALIZE)
```
Comment (henningandersen) [inline x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java:126]: I am not sure I follow this, how do we run into not having enough space available in this test?

**Review by henningandersen — COMMENTED** [review]
I need a little help in understanding this change, see comment.

File: x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java (hunk 1 above)
Comment (fcofdez) [reply x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/SearchableSnapshotsRelocationIntegTests.java:126]: The cache size is set randomly in (https://github.com/elastic/elasticsearch/blob/36683a4581a8fc2f108701d92af2e4d527e56d02/x-pack/plugin/searchable-snapshots/src/internalClusterTest/java/org/elasticsearch/xpack/searchablesnapshots/BaseSearchableSnapshotsIntegTestCase.java#L53) and in some cases it's too small to hold all the data.

**Review by henningandersen — APPROVED** [review]
LGTM.

**Pull request merged**